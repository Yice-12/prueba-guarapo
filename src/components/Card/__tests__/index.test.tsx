import { fireEvent, render } from "@testing-library/react";
import { Card } from "..";
import { BrowserRouter } from "react-router-dom";

const mockCharacter = {
  id: 1,
  name: "Rick Sanchez",
  status: "Alive",
  species: "Human",
  type: "",
  gender: "Male",
  origin: {
    name: "Earth (C-137)",
    url: "https://rickandmortyapi.com/api/location/1",
  },
  location: {
    name: "Citadel of Ricks",
    url: "https://rickandmortyapi.com/api/location/3",
  },
  image: "https://rickandmortyapi.com/api/character/avatar/1.jpeg",
  episode: ["https://rickandmortyapi.com/api/episode/1"],
  url: "https://rickandmortyapi.com/api/character/1",
  created: "2017-11-04T18:48:46.250Z",
};

test("render card and click", () => {
  const { asFragment, getByText } = render(
    <BrowserRouter>
      <Card character={mockCharacter} />
    </BrowserRouter>
  );

  expect(asFragment()).toMatchSnapshot();

  const button = getByText("Rick Sanchez");
  fireEvent.click(button);
});
