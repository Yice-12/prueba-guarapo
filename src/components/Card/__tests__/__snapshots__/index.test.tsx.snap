// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render card 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <div
      class="card__content"
    >
      <img
        src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
      />
      <div
        class="card__description"
      >
        <h1>
          Rick Sanchez
        </h1>
        <p>
          Human
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`render card and click 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <div
      class="card__content"
    >
      <img
        src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
      />
      <div
        class="card__description"
      >
        <h1>
          Rick Sanchez
        </h1>
        <p>
          Human
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
